core/unix/signal_linux_arm.c

```diff
@@ -109,8 +130,38 @@ signal_frame_extra_size(bool include_alignment)
     return 0;
 }
 
+static void
+vfp_query_signal_handler(int sig, siginfo_t *siginfo, kernel_ucontext_t *ucxt)
+{
+    uint offset = 160; /* size of IWMMXT frame */
+    char *coproc = (char *)&ucxt->coproc;
+    kernel_vfp_sigframe_t *vfp0 = (kernel_vfp_sigframe_t *)coproc;
+    kernel_vfp_sigframe_t *vfp1 = (kernel_vfp_sigframe_t *)(coproc + offset);
+    bool vfp0_good = (vfp0->magic == VFP_MAGIC &&
```

Comment:
IMHO there needs to be a comment in this routine so we know what it's doing: pointing at the comment several functions above would be sufficient.